src/plots/cartesian/layout_attributes.js

```diff
@@ -248,6 +248,135 @@ module.exports = {
             'Moreover, note that matching axes must have the same `type`.'
         ].join(' ')
     },
+
+    breaks: templatedArray('break', {
+        enabled: {
+            valType: 'boolean',
+            role: 'info',
+            dflt: true,
+            editType: 'calc',
+            description: [
+                'Determines whether this axis break is enabled or disabled.'
+            ].join(' ')
+        },
+
+        bounds: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'any', editType: 'calc'},
+                {valType: 'any', editType: 'calc'}
+            ],
+            editType: 'calc',
+            description: [
+                'Sets the lower and upper bounds of this axis break.',
+                'Can be used with `operation` to determine the behavior at the bounds.',
+                'On *date* axes, it can be used `pattern`.'
+            ].join(' ')
+        },
+
+        pattern: {
+            valType: 'enumerated',
+            // TODO could add '%H:%M:%S'
+            values: ['%w', '%H', ''],
+            dflt: '',
+            role: 'info',
+            editType: 'calc',
+            description: [
+                'Only coerced on *date* axes.',
+                'Determines a pattern on the time line that generates breaks.',
+                'If *%w* - Sunday-based weekday as a decimal number [0, 6].',
+                'If *%H* - hour (24-hour clock) as a decimal number [0, 23].',
+                'These are the same directive as in `tickformat`, see',
+                'https://github.com/d3/d3-time-format#locale_format',
+                'for more info.',
+                'Examples:',
+                '- { pattern: \'%w\', bounds: [6, 0], operation: \'[]\' }',
+                '  breaks from Saturday to Monday (i.e. skips the weekends).',
+                '- { pattern: \'%H\', bounds: [17, 8] }',
+                '  breaks from 5pm to 8am (i.e. skips non-work hours).'
+            ].join(' ')
+        },
+
+        values: {
+            valType: 'info_array',
+            freeLength: true,
+            role: 'info',
+            editType: 'calc',
+            items: {
+                valType: 'any',
+                editType: 'calc'
+            },
+            description: [
+                'Sets the coordinate values corresponding to the breaks.',
+                'An alternative to `bounds`.',
+                'Use `dvalue` to set the spread of the values along the axis.'
+            ].join(' ')
+        },
+        dvalue: {
+            // TODO could become 'any' to add support for 'months', 'years'
+            valType: 'number',
+            role: 'info',
+            editType: 'calc',
+            min: 0,
+            description: [
+                'Sets the spread of each `values` item.',
```

Comment:
I don't really understand what this means... "spread" is unclear in this context, can you clarify please?

Reply:
An axis break is an continuous interval, so when setting an axis break via a set of values, one needs to also specify the distance on the axis that one value corresponds to.

Reply:
As discussed, I think "size" is less confusing than "spread" and an inline example would help a lot :)

Reply:
Done in https://github.com/plotly/plotly.js/pull/4614/commits/7080f900074a59c6af3fd664999d8493e9d2fe84